src/helpers/socksProxy/getTargetHost.ts

```diff
@@ -0,0 +1,11 @@
+import { checkDomain } from '@/helpers/domain';
+import { ProxyDetails } from '@/helpers/socksProxy/socksProxy.types';
+
+export const getTargetHost = (host: string, proxyDetails: Record<string, ProxyDetails>) => {
+  if (proxyDetails[host]) {
```

Comment:
Safer.

```suggestion
  if (Object.hasOwn(proxyDetails, host)) {
```

Reply:
Thanks, I will not add it as part of this PR. But will evaluate the whole code base use of that pattern in the future.